msm-sitemap.php

```diff
@@ -441,7 +441,7 @@ public static function generate_sitemap_for_date( $sitemap_date ) {
 
 			$url = $xml->addChild( 'url' );
 			$url->addChild( 'loc', get_permalink() );
-			$url->addChild( 'lastmod', get_the_modified_date( 'Y-m-d' ) . 'T' . get_the_modified_date( 'H:i:s' ) . 'Z' );
+			$url->addChild( 'lastmod', get_post_modified_time( 'Y-m-dTH:i:sZ', true ) );
```

Comment:
@romancandlethoughts found a bug, and I think we've tracked it to this change -- would it be possible to use `c` format here? Or if not, the `T` needs to be escaped I think, and possibly the `Z` swapped for `P`? 

I'd be happy to make a PR if you'd like, but figured it'd make sense to start a discussion first. Thanks!